### ISSUE TITLE
onCreateUser not called when inserting a new user

### ISSUE THREAD

**Issue by elie222** [spencercarli/meteor-accounts-google-oauth#1]
Hi,

My app has a `firstName` and `lastName` field in the user's `profile`. These are set in my app by `Accounts.onCreateUser((options, user) => {...}`.

This works with the regular `accounts-google` package, but not with this package, since this package tries to insert a user directly and `onCreateUser` isn't called.

My `onCreateUser` actually does a bunch more things and it's a bit problematic that it isn't called when using this package. I can come up with a hacky solution by removing this package and copying and pasting the code into my app and changing it to my needs, but it would be nice if we could come up with a real solution that works the same way as the other Meteor accounts-x packages.

The same problem exists with the RN Facebook snippet you posted on Medium.

**Comment by elie222** [comment]
From looking at the Meteor code, I can see that only `accounts-password` and `accounts-oauth` packages call the `registerLoginHandler` method, so this is seemingly not the way to do this.

**Comment by elie222** [comment]
Okay, so I've been looking around and thinking about this for a few hours straight now. 4am where I am.

But shouldn't we be able to get everything working if we just recreate the OAuth client side code in React?
Basically just taking these files, and modifying them to work with React:
https://github.com/meteor/meteor/blob/master/packages/accounts-oauth/oauth_client.js
https://github.com/meteor/meteor/blob/master/packages/accounts-oauth/oauth_common.js

Surely can't be too much work? Less than 100 lines of code there.

Unless there's a reason this won't work (which may well be the case and 4am brain is way off), I'm going to give it a try tomorrow.

Hm... just realised this package also relies on other packages, so it's more than 100 lines we need to convert. These are the direct dependencies:

```
  api.use('underscore', ['client', 'server']);
  api.use('random', ['client', 'server']);
  api.use('check', ['client', 'server']);
  api.use('webapp', 'server');
  api.use('accounts-base', ['client', 'server']);
```

Still hoping it isn't too much work to rewrite that code to work with RNMeteor.

**Comment by spencercarli** [comment]
Hey! Sorry this package is causing you pain! I haven't looked at it in quite a while (and don't have any projects actively using it) but calling [`Meteor.users.insert`](https://github.com/spencercarli/meteor-accounts-google-oauth/blob/master/google.js#L71) definitely doesn't seem correct.

I think it would probably be best to call Accounts.createUser with the email address we get back from the request and then updating that user document with all of the other data we want - does that make sense?

**Comment by elie222** [comment]
Yeah. In the Meteor docs it says not to use Accounts.createUser for social
login, but maybe the approach you mentioned above would still work.

For now I've just copied and pasted the package code and made sure the code
that is running in my onCreateUser function also happen when someone does a
Google signup.

**Comment by infacq** [comment]
have you consider this

https://github.com/meteor/meteor/blob/devel/packages/accounts-base/accounts_server.js#L1129

**Comment by mgscreativa** [comment]
Maybe this from google-oauth package:

```
  return Accounts.updateOrCreateUserFromExternalService("google", {
    id: request.userId,
    idToken: request.idToken,
    accessToken: request.accessToken,
    email: request.email,
    picture: request.imageUrl,
    ...result.serviceData,
  }, result.options);
```
